funcx_sdk/funcx/sdk/container_spec.py

```diff
@@ -1,18 +1,52 @@
+import re
+from typing import List, Optional
+
+
 class ContainerSpec:
+    # RegEx copied from https://ihateregex.io/expr/semver/
+    semver_regex = re.compile(
+        r"^(0|[1-9]\d*)\.(0|[1-9]\d*)(\.(0|[1-9]\d*))?(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$"  # noqa E501
+    )
+
     def __init__(
         self,
-        name=None,
-        description=None,
-        apt=None,
-        pip=None,
-        conda=None,
-        payload_url=None,
+        name: Optional[str] = None,
+        description: Optional[str] = None,
+        apt: Optional[List[str]] = None,
+        pip: Optional[List[str]] = None,
+        conda: Optional[List[str]] = None,
+        python_version: str = "3.7",  # This is the default for repo2docker
+        payload_url: Optional[str] = None,
     ):
+        """
+        Construct a container spec to pass to service for build operation.
+        :param name: Name of this container to be used inside funcx
+        :param description: Description of the container inside funcx
+        :param apt: List of Ubuntu library packages to install in container
+        :param pip: List of python libraries to install from pypi
+        :param conda: List of conda packages to install
+        :param payload_url: GitHub repo or publicly readable zip file into copy into container # noqa E501
+        """
+        if not self.semver_regex.match(python_version):
+            raise ValueError("Python version must be a valid semantic version")
+
         self.name = name
         self.description = description
         self.apt = apt if apt else []
         self.pip = pip if pip else []
         self.conda = conda if conda else []
+
+        # Specify the python version in the conda environment. Don't mess with a
+        # user-provided setting in their conda environment if there
+        python_in_env = False
+        for conda_setting in self.conda:
+            if conda_setting.startswith("python="):
+                python_in_env = True
+                break
```

Comment:
Another option might be to inline this:
```python
python_in_env = any(s for s in self.conda if s.startswith("python="))
```
Arguably simpler to read, presuming some Python familiarity.